bigchaindb/commands/bigchaindb.py

```diff
@@ -124,24 +126,32 @@ def run_upsert_validator_new(args, bigchain):
     """
 
     new_validator = {
-        'public_key': args.public_key,
+        'public_key': public_key_from_base64(args.public_key),
         'power': args.power,
         'node_id': args.node_id
     }
 
-    key = load_node_key(args.sk)
-
-    voters = ValidatorElection.recipients(bigchain)
+    try:
+        key = load_node_key(args.sk)
+        voters = ValidatorElection.recipients(bigchain)
+        election = ValidatorElection.generate([key.public_key],
+                                              voters,
+                                              new_validator, None).sign([key.private_key])
+        election.validate(bigchain)
+    except ValidationError as e:
+        logger.error(e)
+        return False
```

Comment:
Could you update the docstring as well (both of `new` and `approve`)?

Reply:
Updated the docstring